app/services/fetch_projects.rb

```diff
@@ -42,9 +42,9 @@ def call
     team_member
   end
 
-  private def create_projects(assignment, team_member)
     puts "Assignable id: #{assignment.assignable_id}"
 
+  private def create_assignment(assignment, team_member)
     assignable_project = projects[assignment.assignable_id]
```

Comment:
Thanks for the feedback! I'm going to schedule revisiting this branch for Tuesday, because Friday is the day of ALL the meetings, and I have another project too long neglected that I want to prioritise.

Reply:
> It looks like the language of "projects" and "assignments" are used quite a bit inside this method. I would be interested in learning more about the thought process around this choice.

Do you have any specific concerns about it?

I can only speculate, based on the commits where they were introduced:
e821fb71fa6d320a9da8e20b7c39086177bcc826
65475f63fe17bbbf68a93900a8ef95628364d471

"Project" and "assignment" come from 10kft terminology.

An assignment is what connects a team member with a project, so it looks like it was used as `has_many :through` glue instead of a `has_and_belongs_to_many` association. The commits don't record whether a HABTM was considered at all, but I didn't find any compelling reasons to change the structure.

Reply:
> It looks like this method receives a resource called "assignment" but is now called "create_assignment", this now seems less clear to me.

The `assignment` resource that this method receives is a 10kft assignment, and the `assignment` it creates is a team dashboard ActiveRecord `assignment` entity.

> Additionally when drilling into what the method does it looks to call `create_or_update_project` which creates a `Project` in our database. Given it does this "create_project" as a name seems honest to me.

Team members and projects are only created if a corresponding record can't be found, and updated if any attributes have changed, so this method won't always create a project, but it will always create an assignment, because we need to wipe out all assignments.

The data we receive from 10kft only represents current assignments, so if we only ever add them to the dashboard database, team members will appear to still be assigned to certain projects even if they aren't currently.